dipy/core/gradients.py

```diff
@@ -210,7 +211,26 @@ def bvecs(self):
         denom = self.bvals + (self.bvals == 0)
         denom = denom.reshape((-1, 1))
         return self.gradients / denom
-
+    
+    @deprecated_params('threshold', None, since='1.17', until='1.19')
+    def getitem(self, bmin=0, bmax=np.inf, threshold=0):
```

Comment:
I think, we got lost here.  

The goal was to have the slicing capabilities. However, we loose it from your initial implementation.  
we can not do anymore gtab[100:200].

we should get back to that. we just need a warning if the slicing is really different from the `b0_treshold`.

Sorry for that. Can you revert your code, use `__get_item__`, and in case of a slice, you check if the lower bound is similar to `b0_thresold`. If yes, nothing to do, if no, just update self.b0_thresold to this value.

Does it make sense?   we do not need(bmin, bmax, thresold)

Reply:
Sure, you're right. We did get deviated from the original method. I changed it, Did you mean something like this? If you'd agree with this, i'll change the test cases as well.